### PR TITLE
Stats: add Odyssey Stats to TeamCity build config - Take #2

### PR DESCRIPTION
**Pull request by kangzj**
#### Proposed Changes

This is a follow up for https://github.com/Automattic/wp-calypso/pull/70353. The pluginSlug has to match the directory name in apps. Build error message following [the instructions](https://github.com/Automattic/wp-calypso/pull/70353#issuecomment-1338192868):

```
  /home/teamcity-0/xx/xx/xx/custom_script184942528485828883: line 14: cd: apps/calypso-stats: No such file or directory
```

It changes the directory name to `odyssey-stats` to be consistent and avoid confusion.

#### Testing Instructions

* It has to be merged to be tested - `p58i-dox#comment-56262-p2`, so please check whether there are obvious mistakes.

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by jsnmoon — APPROVED** [review]

**Review by noahtallen — APPROVED** [review]

File: .teamcity/_self/projects/WPComPlugins.kt [hunk 1]
```diff
@@ -29,7 +29,7 @@ object WPComPlugins : Project({
 	buildType(EditingToolkit)
 	buildType(WpcomBlockEditor)
 	buildType(Notifications)
-	buildType(CalypsoStats)
+	buildType(OdysseyStats)
```
Comment (noahtallen) [inline .teamcity/_self/projects/WPComPlugins.kt:32]: Definitely a good time to change the build ID, since it'd be harder to do that at a later time

**Comment by noahtallen** [comment]
Calypso Stats won't succeed in this build since the build ID is changing

File: .teamcity/_self/projects/WPComPlugins.kt (hunk 1 above)
Comment (kangzj) [reply .teamcity/_self/projects/WPComPlugins.kt:32]: @noahtallen thanks for the information 🙏

**Comment by kangzj** [comment]
@noahtallen Thanks heaps for all your help. It worked! 🥳

**Comment by noahtallen** [comment]
You're welcome!

For next steps, you could take a look at this diff: D90725-code. You just have to make the same changes, just replacing most text with what makes sense for stats. (excluding the change in that diff which loads the files, of course!) Then something like `install-plugin.sk odyssey-stats --release` should start working as the deploy workflow :)

**Comment by noahtallen** [comment]
After all that is set up, you probably won't need to make any changes to the deploy pipeline. You can modify the webpack config as much as needed, and as long as everything still outputs to the /dist folder, those changes will ultimately get reflected in the widgets.wp.com directory as well.

**Pull request merged**